### PR TITLE
Add some CI, flake, nixos test

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    - name: Install Nix
+      uses: cachix/install-nix-action@v17
+      with:
+        extra_nix_config: |
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+          experimental-features = nix-command flakes
+    - name: Build
+      run: nix build -L
+    #- name: Check -- not enabled because we need nested virt :(.
+    #  run: nix flake check -L
```
Comment (grahamc) [inline .github/workflows/ci.yml:46]: It'd be cool if this could run without kvm on the host.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+        in
+        {
+          default = pkgs.buildGoModule {
+            pname = "dhcpv6macd";
+            inherit version;
+            # In 'nix develop', we don't need a copy of the source tree
```
Comment (lheckemann) [inline flake.nix:51]: How is `nix develop` relevant here? Are we expecting people to `nix develop .#packages.x86_64-linux.packages.default`?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+
+    in
+    {
+      devShell = forAllSystems
```
Comment (lheckemann) [inline flake.nix:28]: Deprecated in favour of `devShells.${system}.default`. Though I think maybe it should be undeprecated.

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                };
+                ipv6SendRAConfig.Managed = true;
+                linkConfig.RequiredForOnline = false;
+              };
+            };
+          };
+
+          nodes.client = {
+            systemd.network.networks."40-eth1" = {
+              dhcpV6Config.DUIDType = "link-layer";
```
Comment (lheckemann) [inline flake.nix:159]: This shouldn't be necessary, since the server ignores the DUID
```suggestion
```

**Review by lheckemann — APPROVED** [review]

**Pull request merged**